### DISCUSSION TITLE
Handling citations in Readium Web

### DISCUSSION THREAD

**Discussion by HadrienGardeur** [readium/web#8] [General]
Compared to our other projects, Readium Web is even more likely to be deployed by academic publishers and/or libraries where citations are an important feature.

There are many [citation styles](https://en.wikipedia.org/wiki/Citation#Styles) in use, but all of them rely on a mix of:

- metadata about the publication
- and in some cases, information about the chapter/pages that are cited

Getting high quality metadata can be challenging though and there could be issues for example with the format used for author names.

Should we provide this feature somehow in Readium Web and potentially our other projects as well?

I believe that this could be a helper that implementers could call by providing the `metadata` object of an RWPM with their preferred style as a parameter. This way, it could default to `metadata` from the publication itself, but could be overridden when necessary (for example with OPDS metadata).

For additional context, we could include page references (when they're present in the publication) and/or use the `<title>` of the current resource or the table of contents to identify the current chapter/section.

**Comment by JamesEnglish1028** [comment]
In the context of citations and academic use - the helper function cold be usefull for pulling teh citation metadata for display in some type of html presentation, but in likelyhood,  and academic library would like the citation to be exportable to a citation management tool such as Medeley or Zoterro.  In that case, the [CSL-JSON](https://citeproc-js.readthedocs.io/en/latest/csl-json/markup.html) maybe or[ RIS ](https://en.wikipedia.org/wiki/RIS_(file_format))files may be useful to generate from the helper as well so the user can add the the citation into their citation management tool.
